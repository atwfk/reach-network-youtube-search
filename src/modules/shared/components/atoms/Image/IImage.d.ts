declare namespace IImage {
  export interface IProps {
    src: string;
    alt: string;
    width?: string;
    height?: string;
    isChannel?: boolean;
  }
}

export { IImage };
