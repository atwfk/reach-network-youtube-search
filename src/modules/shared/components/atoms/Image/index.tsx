import React from "react";
import type { FC, ReactElement } from "react";
import { IImage } from "./IImage";

const Image: FC<IImage.IProps> = ({ src, alt, width, height }): ReactElement => {
  return <img src={src} alt={alt} width={width} height={height} data-testid="image" />;
};

export default Image;
