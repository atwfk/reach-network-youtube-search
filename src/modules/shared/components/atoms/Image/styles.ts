import styled from "styled-components";

export const StyledImageContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  .channel {
    border-radius: 50%;
  }
`;
