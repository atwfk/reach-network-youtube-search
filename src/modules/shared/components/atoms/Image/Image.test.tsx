import { render, screen, cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import Image from ".";

afterEach(cleanup);

describe("Image Component snapshots", () => {
  it("Should render simple Image component", () => {
    const tree = renderer
      .create(
        <Image
          src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
          alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
          width="320"
          height="180"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("Image Component RTL", () => {
  it("should have image with channel class when we pass isChannel prop", () => {
    render(
      <Image
        src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
        alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
        width="320"
        height="180"
        isChannel
      />
    );

    const image = screen.getByTestId("image");

    expect(image).toHaveClass("channel");
  });

  it("should have image without channel class when we don't pass isChannel prop", () => {
    render(
      <Image
        src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
        alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
        width="320"
        height="180"
      />
    );

    const image = screen.getByTestId("image");

    expect(image).not.toHaveClass("channel");
  });
});
