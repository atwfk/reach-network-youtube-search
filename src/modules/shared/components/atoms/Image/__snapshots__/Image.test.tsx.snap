// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component snapshots Should render simple Image component 1`] = `
<img
  alt="Kubernetes Tutorial for Beginners [FULL COURSE in 4 Hours]"
  data-testid="image"
  height="180"
  loading="lazy"
  src="https://i.ytimg.com/vi/X48VuDVv0do/mqdefault.jpg"
  width="320"
/>
`;
